Gravity/Gravity/DAL/RSAPI/RsapiDao.Insert.cs

```diff
@@ -124,23 +122,88 @@ public void InsertChildListObjects<T>(IList<T> objectsToInserted, int parentArti
 				var rdosToBeInserted = objectsToInserted.Select(x => x.ToRdo()).ToArray();
 
 				rsapiProvider.Create(rdosToBeInserted);
-			}			
+			}
+		}
+
+		private bool InsertUpdateSingleObjectFields(BaseDto objectToInsert)
+		{
+			foreach (var propertyInfo in objectToInsert.GetType().GetProperties().Where(c =>
+				c.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.SingleObject))
```

Comment:
Don't query for the attribute twice. Try and find it, and move on if it is null

Reply:
@Arithmomaniac Not sure what you mean here?  I need to get all the SingleObject field types, then do stuff with those fields.  Suggestions requested for clarity.

Reply:
```csharp
foreach (var propertyInfo in objectToInsert.GetType().GetProperties())
{
	var attribute = propertyInfo.GetCustomAttribute<RelativityObjectFieldAttribute>();
	if (attribute?.FieldType == RdoFieldType.SingleObject)
	{
		var fieldValue = (objectToInsert.GetPropertyValue(propertyInfo.Name) as BaseDto);
		if (fieldValue != null)
		{
			var fieldGuid = attribute.FieldGuid;
			//...
		}
	}
}
```